### PR TITLE
Simran

### PR DESCRIPTION
**Pull request by Simran2711**
Please review code.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+from queries import *
+from issue import *
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+
+# Creating a Flask application instance and enabling CORS (Cross-Origin Resource Sharing) for handling cross-origin requests.
+app = Flask(__name__)
+CORS(app, origins="*")
+
```
Comment (Chandrashekhar-Bhavsar) [inline app.py:14]: Properly indented the code

File: app.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+############################ CREATE ISSUE DETAILS #################################
+
+# Defining an API endpoint (/create_issue) for creating a new issue. This endpoint expects a POST request.
+@app.route('/create_issue', methods=['POST'])
+def create_issue():
+    
+    logging.debug("calling createissue function.")
+    return createissue()  
+
```
Comment (Chandrashekhar-Bhavsar) [inline app.py:23]: Properly indented the code

File: db.config [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "host": "localhost",
```
Comment (Chandrashekhar-Bhavsar) [inline db.config:2]: Change MySQL configuration as per open shift pod

File: issue.py [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+
+def createissue():
+    try:
+        logging.debug("entered into create issue" )
+        data = request.get_json()
+        Issue_name = data['Issue_name']
+        Description = data['Description']
+        logging.debug("accepted values")
+
```
Comment (Chandrashekhar-Bhavsar) [inline issue.py:27]: Properly indented the code

File: issue.py [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+        logging.error("Missing key in request data: " + str(e))
+        return jsonify({"error": "Missing required parameter"+  str(e)}), 400
+
+
+    except Exception as e:
+        # Handle any other unexpected exceptions
+        logging.error("An error occurred: " + str(e))
+        return jsonify({"error": "An error occurred: " + str(e)}), 500
+    
+
```
Comment (Chandrashekhar-Bhavsar) [inline issue.py:88]: Properly indented the code

File: issue.py [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+        logging.debug("calling updatedefects function.")
+        return updatedefects(description, status, severity, defect_sd, defect_ed, planned_hours, actual_hours,defect_id, issue_id)
+
+    except KeyError as e:
+        # Handle missing key in the request data
+        logging.error("Missing key in request data: " + str(e))
+        return jsonify({"error": "Missing key in request data: " + str(e)}), 400
+
+
```
Comment (Chandrashekhar-Bhavsar) [inline issue.py:160]: Properly indented the code

**Review by Chandrashekhar-Bhavsar — CHANGES_REQUESTED** [review]
No unit testing for apis  and Some Formatting needed

File: app.py (hunk 1 above)
Comment (Simran2711) [reply app.py:14]: Yes, I'll do the needful

File: app.py (hunk 2 above)
Comment (Simran2711) [reply app.py:23]: Yes, I'll do the needful

File: issue.py [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+   
+
+    except KeyError as e:
+        # Handle missing required parameter
+        logging.error("Missing key in request data: " + str(e))
+        return jsonify({"error": "Missing required parameter: {}".format(str(e))}), 400
+
+    
+
+############################ DELETE ISSUE DETAILS #################################
```
[the remark this answers is not in the document]
Comment (Simran2711) [reply issue.py:66]: Yes,I will update it

File: issue.py (hunk 5 above)
Comment (Simran2711) [reply issue.py:79]: All the said changes are completed

**Review by Chandrashekhar-Bhavsar — APPROVED** [review]

**Review by pratiksatham — APPROVED** [review]

**Pull request closed without merging**